frontend/src/pages/components/displayImages.js

```diff
@@ -836,10 +836,46 @@ function ShareIcon() {
 
 
 /* Save Icon */
-function SaveIcon() {
+function SaveIcon(props) {
+
+  const newProfilePic = props.code;
+
+  function changeProfilePic() {
+    //grab the user's id
+    fetch('/api?action=getUser', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      credentials: 'include'
+    })
+      .then(res => res.json())
+      .then(user => {
+        if (user) {
+
+          // change the user's profile picture
+          fetch("/api", {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            credentials: "include",
+            body: JSON.stringify({
+              action: "changeProfilePic",
+              newProfilePic: newProfilePic,
+              id: user._id,
+            }),
+          })
+            .then((res) => res.json())
+            .then((message) => alert(message));
+        }
+      })
+  }
```

Comment:
It looks like the first step here is that we ask the server what the is requestor's user id. I believe that this eventually gets to database.changeProfilePic, within which expects id in the body. However, from observing the other other API handlers, the id is already stored in req.user._id. The extra HTTP request is unnecessary and would slow down the site, even more so if the user document is particularly large. I would suggest modifying either of the changeProfilePic API handler or changeProfilePic database function to retrieve the id from req.user._id rather than from the request body as this would allow us to eliminate the first fetch request which retrieves a user document _just_ to use the id.

Reply:
I did not know we already had it! That's good to know.